Sources/HaCWebsiteLib/Events/HackathonEvent.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+struct HackathonEvent : Event {
+    let title : String
+    let time : Date 
+    let eventDescription: String
+    let colour : String
+    let hypeDays : Int
+    let ttlDays : Int
+    let duration : Int
+    var sponsor : String?
```

Comment:
Do these need to be mutable?

Reply:
So they were left mutable so that they could be optionally set but not in the constructor. I wasn't sure if it was possible to do this with immutables. I could add them to the constructor but I was worried about flexibility??